Descent3/aipath.cpp

```diff
@@ -651,6 +651,8 @@ static inline bool AIPathAddDPathNode(ai_path_info *aip, int *slot, int *cur_nod
     status = AIPathAddDPath(aip, handle);
     if (!status)
       Int3(); // chrishack -- out of dynamic paths
+    if (aip->num_paths < 0)
```

Comment:
below we have `aip->path_id[aip->num_paths - 1]`, so we should return if `aip->num_paths < 1` instead. Same logic below